pmlc/dialect/pxa/transforms/autotile.cc

```diff
@@ -0,0 +1,29 @@
+// Copyright 2020, Intel Corporation
+
+#include "pmlc/dialect/pxa/transforms/autotile.h"
+
+namespace pmlc::dialect::pxa {
+
+std::vector<int64_t> PowerOfTwoGenerator(int64_t range) {
+  std::vector<int64_t> out;
+  for (int64_t r = 1; r <= range; r *= 2) {
+    out.push_back(r);
+  }
+  return out;
+}
+
+std::vector<int64_t> EvenDivisionGenerator(int64_t range) {
+  std::vector<int64_t> out;
+  // TODO: Something less naive: i.e. factor with sieve and then produce
+  // divisors via that.  This is not as bad as one might imagine, since generator
+  // set is cached in autotile.
+  for (int64_t r = 1; r <= range; r++) {
+    if (range % r != 0) {
```

Comment:
I think there's one trivial optimization you could do here -- just scan up to the sqrt of the range, and add the paired divisors as you go?

Reply:
We don't want prime factors, we want all the divisors (including the full range).  So for 16, we want 1, 2, 4, 8, 16, and sqrt of 16 is only 4.